File: backend/app/models/transaction.rb
class Transaction < ApplicationRecord
  validates :reference, presence: true, uniqueness: true
  validates :flow_type, presence: true
  validates :status, presence: true
  validates :amount, presence: true, numericality: { greater_than: 0 }
  validates :currency, presence: true
  validates :country, presence: true
end

File: frontend/src/types/transaction.ts
export interface Transaction {
  id: number;
  reference: string;
  flow_type: string;
  status: string;
  amount: string;
  currency: string;
  country: string;
  customer_name: string;
  beneficiary_name: string;
  masked_account: string;
  failure_reason: string | null;
  created_at: string;
  updated_at: string;
}